File: Problems/Algorithms/1790. Check If One String Swap Can Make Strings Equal/StringSwap.kt
class Solution {
    fun areAlmostEqual(s1: String, s2: String): Boolean {
        var count = 0
        var c1 = '.'
        var c2 = '.'
        for (i in 0..s1.length-1) {
            if (s1.get(i) != s2.get(i)) {
                if (count == 2) {
                    return false
                } else if (count == 0) {
                    c1 = s1.get(i)
                    c2 = s2.get(i)
                } else {
                    if (c1 != s2.get(i) || c2 != s1.get(i)) return false
                }
                count++
            }
        }
        
        if (count == 0 || count == 2) return true
        return false
    }
}

File: Problems/Algorithms/1790. Check If One String Swap Can Make Strings Equal/string-swap.cpp
#include <string>

using namespace std;

class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        int first = 0, second = 0;
        int diffs = 0;

        for (int i = 0; i < s1.size(); i++) {
            if (s1[i] != s2[i]) {
                diffs++;

                if (diffs > 2) {
                    return false;
                } else if (diffs == 1) {
                    first = i;
                } else {
                    second = i;
                }
            }
        }

        return s1[first] == s2[second] && s1[second] == s2[first];
    }
};
